### DISCUSSION TITLE
How can I use .find() with await ?

### DISCUSSION THREAD

**Discussion by Redemption007** [Automattic/mongoose#10930] [Q&A] [answered]
Hello,
I've seen the FAQ, and I would like to learn more about [this one](https://github.com/Automattic/mongoose/blob/master/docs/faq.md#duplicate-query).
Indeed, I don't want to use Promises in my code, so I was using the async/await solution.
```js
const reactors = await Reactor.find(r => {
  if (!r) return null
  return r.channelID === oldchannel.id
})
```
As expected, console throws `MongooseError: Query was already executed: Reactor.find({})`
> According to the FAQ, I have to remove the await.

But then my code doesn't wait the end of the research, while I'm using `reactors` just after in my file.
Do you have any idea how I can resolve this problem ?
Thank you in advance

**Comment by vkarpov15** [comment, marked as answer]
You're using `find()` incorrectly. If you call `find()` with a function, Mongoose assumes the function is a callback to execute when the query completes, **not** a function that filters documents. You should do `await Reactor.find({ channelID: oldchannel.id })`

**Reply by Redemption007** [reply]
Thanks a lot !!!
